File: SayiTahmini/SayiTahmini.java
package sayitahmini;

import java.util.Scanner;

public class SayiTahmini {

    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        int bilgisayar = (int) (Math.random() * 10 + 1);
        System.out.print("Bir tamsayı giriniz(1-10): ");
        int adim = 0;
        for (;;) {
            int oyuncu = reader.nextInt();
            if (oyuncu > bilgisayar) {
                System.out.print("Daha küçük sayı girin: ");
                adim += 1;
            } else if (oyuncu < bilgisayar) {
                System.out.print("Daha büyük sayı girin: ");
                adim += 1;
            } else if (oyuncu == bilgisayar) {
                adim += 1;
                System.out.println(adim + " ADIMDA BULDUNUZ");
                break;
            }
        }
    }
}

File: SayiTahmini/SayiTahmini.py
import random


def sayi_tahmin_oyunu():

    bilgisayar = random.randint(1, 10)
    adim_sayisi = 0

    while True:

        kullanici = int(input("Bir sayı tahmin edin : "))

        if kullanici == bilgisayar:
            if adim_sayisi == 0:
                print("Tebrikler doğru tahmin\nilk adımda buldun")
            else:
                print("Tebrikler doğru tahmin\n{} adımda buldun".format(adim_sayisi+1))
                break
        elif kullanici > bilgisayar:
            print("Daha küçük bir sayi girin : ")
            adim_sayisi += 1
        else:
            print("Daha büyük bir sayi girin : ")
            adim_sayisi += 1


sayi_tahmin_oyunu()